### PR TITLE
fix: drag and extend

### PR DESCRIPTION
**Pull request by thetaungg**

### CODE REVIEW

File: src/datagrid/plugins/drag-and-extend.tsx [hunk 1]
```diff
@@ -155,10 +155,10 @@ const useOnDragSelection = ({
 
         // This means it's horizontal selection
         if (rowHovered && checkIfWithinRow(cellsSelection, { rowIndex })) {
-            const startColumnIndexOffset = columnIndex > cellsSelection.end.columnIndex ? 1 : -1;
-            const startColumnIndex = cellsSelection.end.columnIndex + startColumnIndexOffset;
-            const startRowIndex = cellsSelection.start.rowIndex;
-            const endRowIndex = cellsSelection.end.rowIndex;
+            const startColumnIndexOffset = columnIndex > cellsSelection.end?.columnIndex ? 1 : -1;
+            const startColumnIndex = cellsSelection.end?.columnIndex + startColumnIndexOffset;
```
Comment (elvisduru) [inline src/datagrid/plugins/drag-and-extend.tsx:159]: If undefined, result will be NaN

File: src/datagrid/plugins/drag-and-extend.tsx [hunk 2]
```diff
@@ -184,11 +184,11 @@ const useOnDragSelection = ({
 
         // This means it's vertical selection
         if (rowHovered && !checkIfWithinRow(cellsSelection, { rowIndex })) {
-            const startRowIndexOffset = rowIndex > cellsSelection.end.rowIndex ? 1 : -1;
-            const startColumnIndex = cellsSelection.start.columnIndex;
-            const startRowIndex = cellsSelection.end.rowIndex + startRowIndexOffset;
+            const startRowIndexOffset = rowIndex > cellsSelection.end?.rowIndex ? 1 : -1;
+            const startColumnIndex = cellsSelection.start?.columnIndex;
+            const startRowIndex = cellsSelection.end?.rowIndex + startRowIndexOffset;
```
Comment (elvisduru) [inline src/datagrid/plugins/drag-and-extend.tsx:189]: same

File: src/datagrid/plugins/drag-and-extend.tsx (hunk 1 above)
Comment (thetaungg) [reply src/datagrid/plugins/drag-and-extend.tsx:159]: It's fine. I have an if statement below to make sure they exist.
It won't never be undefined anyways. This is just a safe guard

Comment (elvisduru) [reply src/datagrid/plugins/drag-and-extend.tsx:159]: Alright bro

**Review by elvisduru — APPROVED** [review]

**Pull request merged**